frontend/dockerfile/parser/parser.go

```diff
@@ -275,6 +275,11 @@ func Parse(rwc io.Reader) (*Result, error) {
 	if len(warnings) > 0 {
 		warnings = append(warnings, "[WARNING]: Empty continuation lines will become errors in a future release.")
 	}
+
+	if root.StartLine < 0 {
+		return nil, errors.New("Dockerfile contains no instructions.")
```

Comment:
nit: leave this lowercase to support wrapping. eg "empty file with no instructions".